arrows/vxl/hashed_image_classifier.h

```diff
@@ -0,0 +1,191 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#ifndef HASHED_IMAGE_CLASSIFIER_H_
+#define HASHED_IMAGE_CLASSIFIER_H_
+
+#include <vgl/vgl_box_2d.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_rgb.h>
+
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <vector>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+template < typename FeatureType,
+           typename OutputType > class hashed_image_classifier;
+
+/// Internal data required for the hashed_image_classifier class.
+///
+/// The data members of this class are exposed just in case any external
+/// functions or processes want to train their own models, usually in an
+/// online fashion.
+template < typename FloatType = double >
+class hashed_image_classifier_model
+{
+public:
+  using self_t = hashed_image_classifier_model< FloatType >;
+  using weight_t = FloatType;
+
+  hashed_image_classifier_model() : num_features( 0 ) {}
```

Comment:
```suggestion
  hashed_image_classifier_model() = default;
```
...with later change to inline member initializer.